### PR TITLE
#14: Exclude tests from compilation

### PR DESCRIPTION
**Pull request by Agreon**

### CODE REVIEW

**Comment by angrykoala** [comment]
Thanks for your contribution @Agreon 

Please, could you make the PR to `dev` branch instead of `master?
Thanks

File: tsconfig.json [hunk 1]
```diff
@@ -15,5 +15,6 @@
     "esModuleInterop": true,
     "experimentalDecorators": true
   },
-  "compileOnSave": true
+  "compileOnSave": true,
+  "include": ["./src/**/*.ts"]
```
Comment (angrykoala) [inline tsconfig.json]: this setting won't include index.ts, which is required for the build to work

File: package.json [hunk 2]
```diff
@@ -13,12 +13,13 @@
...
     "tslint": "^5.18.0",
     "tslint-config-prettier": "^1.18.0",
     "typescript": "^3.6.3"
   },
   "scripts": {
-    "test": "npm run tsc && nyc mocha dist/tests/*.test.js",
+    "test": "nyc mocha --require ts-node/register ./tests/*.test.ts",
     "tsc": "rm -rf dist && tsc",
     "tslint": "tslint 'src/**/*.ts?(x)'",
     "prepublishOnly": "npm run tslint && npm test"
```
Comment (angrykoala) [inline package.json]: Because `npm test` no longer execute `tsc`, prepublishOnly should run it after tslint and tests are OK

**Review by angrykoala — CHANGES_REQUESTED** [review]
Would it be worth it to add a new step `npm run tsc` in .travis.yml to ensure the build is working in the CI?

**Comment by Agreon** [comment]
Sorry i must have overlooked the index.ts.

I don't think it is worth it, as it would not fail if the index.ts is missing for example. Otherwise i would have noticed.

**Pull request merged**